jbpm-designer-backend/src/test/java/org/jbpm/designer/editorhandler/EditorHandlerBaseTest.java

```diff
@@ -240,6 +249,57 @@ public void testServiceRepoAndTaskEmpty() throws Exception {
                                   false);
     }
 
+    @Test
+    public void testStoreSVGOnSave() throws Exception {
+        when(editorHandler.getProfile()).thenReturn(mockProfile);
+
+        // default the profile svg on save to true
+        when(mockProfile.getStoreSVGonSaveOption()).thenReturn("true");
+
+        TestServletConfig config = new TestServletConfig(new TestServletContext(repository));
+        PrintWriter writer = mock(PrintWriter.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
+        when(response.getWriter()).thenReturn(writer);
+
+        // default from profile setting, no system prop set
+        editorHandler.init(config);
+        editorHandler.doGet(mock(HttpServletRequest.class),
+                            response);
+        assertEquals("true",
+                     editorHandler.getEditorTemplate().getAttribute("storesvgonsave"));
+
+        // overwrite with system prop
+        System.setProperty(EditorHandler.STORE_SVG_ON_SAVE,
+                           "false");
+        editorHandler.init(config);
+        editorHandler.doGet(mock(HttpServletRequest.class),
+                            response);
+        assertEquals("false",
+                     editorHandler.getEditorTemplate().getAttribute("storesvgonsave"));
+
+        // set profile store on svg to false and overwrite with system prop true
+        when(mockProfile.getStoreSVGonSaveOption()).thenReturn("false");
+        System.setProperty(EditorHandler.STORE_SVG_ON_SAVE,
+                           "true");
+        editorHandler.init(config);
+        editorHandler.doGet(mock(HttpServletRequest.class),
+                            response);
+        assertEquals("true",
+                     editorHandler.getEditorTemplate().getAttribute("storesvgonsave"));
+
+        //overwrite with incorrect system prop  -- should be false
```

Comment:
to not compensate user error. if user decides to start appserver with system prop set but uses a wrong value it will be evaluated to false